src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/code/CodeBlob.java

```diff
@@ -86,6 +115,37 @@ public void update(Observable o, Object data) {
     });
   }
 
+  public static Class<?> getClassFor(Address addr) {
```

Comment:
Did you consider using a lookup table here that is indexed using the kind value?

Reply:
Example please.

Reply:
```
    static Class wrapperClasses = new Class[Number_Of_Kinds];
    wrapperClasses[NMethodKind] = NMethodBlob.class;
    wrapperClasses[BufferKind] = BufferBopb.class;
    ...;
    wrapperClasses[SafepointKind] = SafepointBlob.class;
```

```
    CodeBlob cb = new CodeBlob(addr);
    return wrapperClasses[cb.getKind()];
```

Reply:
Done.